src/test/java/ca/uhn/fhir/jpa/starter/interceptor/CustomSearchNarrowingInterceptorTest.java

```diff
@@ -0,0 +1,205 @@
+import org.hl7.fhir.r4.model.Observation;
+import org.hl7.fhir.r4.model.Patient;
+import org.hl7.fhir.r4.model.Resource;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mockito.InjectMocks;
+import org.mockito.MockedStatic;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.jpa.starter.AppProperties;
+import ca.uhn.fhir.jpa.starter.AppProperties.Oauth;
+import ca.uhn.fhir.jpa.starter.util.OAuth2Helper;
+import ca.uhn.fhir.model.api.IQueryParameterOr;
+import ca.uhn.fhir.model.api.IQueryParameterType;
+import ca.uhn.fhir.rest.annotation.OptionalParam;
+import ca.uhn.fhir.rest.annotation.Search;
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+import ca.uhn.fhir.rest.param.BaseAndListParam;
+import ca.uhn.fhir.rest.param.ReferenceAndListParam;
+import ca.uhn.fhir.rest.param.TokenAndListParam;
+import ca.uhn.fhir.rest.server.IResourceProvider;
+import ca.uhn.fhir.rest.server.interceptor.auth.AuthorizedList;
+import ca.uhn.fhir.test.utilities.server.RestfulServerExtension;
+
+class CustomSearchNarrowingInterceptorTest {
+
+	@Spy
+	AppProperties mockConfig;
+
+	@Spy
+	Oauth mockOAuth;
+
+	@InjectMocks
+	CustomSearchNarrowingInterceptor ourSearchNarrowingInterceptor;
+
+	private static Logger logger = LoggerFactory.getLogger(CustomSearchNarrowingInterceptorTest.class);
+	private static List<Resource> ourReturn;
+	private static String ourLastHitMethod;
+	private static TokenAndListParam ourLastIdParam;
+	private static ReferenceAndListParam ourLastPatientParam;
+	private static final FhirContext ourCtx = FhirContext.forR4();
+	private static final String PATIENT_ID = "12345";
+	public IGenericClient myClient;
+
+	@RegisterExtension
+	public RestfulServerExtension myRestfulServerExtension = new RestfulServerExtension(ourCtx)
+			.registerProvider(new MockPatientResourceProvider())
+			.registerProvider(new MockObservationResourceProvider());
+
+	@BeforeEach
+	public void before() throws IOException {
+		MockitoAnnotations.openMocks(this);
+		ourReturn = Collections.emptyList();
+		ourLastHitMethod = null;
+		ourLastIdParam = null;
+		ourLastPatientParam = null;
+		mockOAuth.setEnabled(false);
+		mockConfig.setOauth(mockOAuth);
+		myClient = myRestfulServerExtension.getFhirClient();
+	}
+
+	@Test
+	void testSearchNarrowObservationWithPatientClaim() throws Exception {
+		mockOAuth.setEnabled(true);
+		ourSearchNarrowingInterceptor = new CustomSearchNarrowingInterceptor(mockConfig) {
+			@Override
+			protected AuthorizedList buildAuthorizedList(RequestDetails theRequestDetails) {
+				try (MockedStatic<OAuth2Helper> mockedStatic = mockStatic(OAuth2Helper.class)) {
+					mockedStatic.when(() -> OAuth2Helper.hasToken(any())).thenReturn(true);
```

Comment:
Mocking, and replacing, the functionality of the OAuth2Helper invalidates the entire test. Tests SHOULD NOT know about the helper.